src/lib/shared/ui/snackbar/store.ts

```diff
@@ -0,0 +1,45 @@
+import { writable, get } from 'svelte/store';
+import type { SnackbarOptions } from './types';
+
+/**
+ * The queue of different snackbars. It works like this:
+ *
+ * - First comes served – the first snackbar stays on the screen for 4 seconds
+ * - Next snackbar waits in the queue for its turn
+ * - Snackbar timing can be prolonged – if the time of the first snackbar in the queue
+ * didn't pass but the call for this type of snackbar fired again, timeout resets
+ * to 4 seconds
+ * - Pressing on the Snackbar's action removes it from the queue
+ */
+export const snackbarQueue = writable<SnackbarOptions[]>([]);
+
+const SNACKBAR_TIMEOUT = 4000;
```

Comment:
Why isn't this configurable for each snackbar? We could have it as part of the options but with a default value

Reply:
I don't think there's a need for this, we're not building a UI library